### ISSUE TITLE
Support for Win7/8?

### ISSUE THREAD

**Issue by LexaGV** [VSCodium/vscodium#1264]
I use Windows7 x64 and VSCodium WAS working on this platform. Why it doesn't work anymore? Is it possible to build Electron for Win7? Obviously there is NOTHING what Electron needs from Win10 what doesn't exist on Win7.

If it's not possible to recompile Electron, it could be nice AT LEAST keep mention of latest release which worked on Win7. Please, update README.

**Comment by africalimedrop** [comment]
See https://github.com/VSCodium/vscodium/issues/1256
They need to do more than 'update the readme', they need to not allow the update on Win 7. We don't need our environments seamlessly fucked.

**Comment by paulcarroty** [comment]
Waiting for https://github.com/electron/electron/issues/35597.

**Comment by daiyam** [comment]
I tried to find a doc for the system requirements to run electron (not to build it). I didn't found it...

**Comment by zedayseekay** [comment]
> 

Windows 7 support is mentioned in https://github.com/electron/electron/blob/main/README.md

Under https://github.com/electron/electron/issues/35597 I have now added a question on whether there is a more formal support matrix - I couldn't find one either, you're right! It doesn't matter in this case because https://github.com/electron/electron/issues/35597 has been treated as valid anyway.

**Comment by LexaGV** [comment]
It's even better news!! If Electron can (and must) run on Win7, then Codium has no problem at all to work on Win7 too! Am I right?

**Comment by GitMensch** [comment]
Note that ["Visual Studio Code" 1.71+ _explicit_ does not work with Windows 7 any more](https://code.visualstudio.com/docs/supporting/faq#_can-i-run-vs-code-on-windows-7), because
> Microsoft [ended support for Windows 7](https://docs.microsoft.com/lifecycle/products/windows-7) in January, 2020 and no longer provides security updates.

If you need an installation now it is likely best to use https://github.com/VSCodium/vscodium/releases/tag/1.70.2.22230 while the upstream Electron issue is solved - and ignore any updates until this issue is closed and a follow-up release is finished.

**Comment by musadarj** [comment]
So the latest working version for Win 7 (for now) is [1.70.2.22230](https://github.com/VSCodium/vscodium/releases/tag/1.70.2.22230), well I'll note that for future reference.

**Comment by LexaGV** [comment]
Guys, thanks for answer. Will Win7 info be reflected in README please? My English is far from good to do it. :)

Related "MS ended support for Win7" - sorry, how it corresponds to development?! Does your operating system gone away from SSD? Or Windows become rusty? Not. SYSTEM WORKS AS BEFORE, so please don't write stupid reasons "MS doesn't support" - nobody care of it, system works as usual and nothing prevents you from using Win7 (as I do now).

**Comment by GitMensch** [comment]
... as noted in the other issue: the README says what is the current state, once the upstream Electron issue is solved there is a _chance_ that Win7 works again.
... and how MS' take corresponds to development: they explicit made clear that they won't take any effort of supporting "Visual Studio Code" under Win 7, and as vscode is Microsoft Copyright'ed and this repo builds the underlying "vscode" where only Microsoft has commit rights the decisions of MS have quite an effect on VSCodium, too.
BTW: VSCodium already provides binaries for more systems than "Visual Studio Code" does; and I personally think that this will also extend to Win 7.
Let's wait on Electron changes, and keep calm until then. 
And of course: you are invited to help them out and once this is solved are invited to also build VSCodium on your Win7 box using this repo's build scripts and patches along with the changed version of Electron and report "first hand" - or wait a bit longer.

**Comment by GitMensch** [comment]
https://github.com/electron/electron/releases/tag/v19.1.0 is out with fixed Win7 support - canb we trigger an insider build that uses this version?

**Comment by GitMensch** [comment]
Ping @daiyam (I guess the published 1.72 does not use the new electron release, does it?)

**Comment by daiyam** [comment]
The `1.72.0` is using electron `v19.0.17`. No fix Win7, yet.

**Comment by GitMensch** [comment]
Thanks for the quick answer. Would it be possible/reasonable to adjust this in the insider version - even if MS has not done it yet?

**Comment by daiyam** [comment]
I think it's possible.

**Comment by AXKuhta** [comment]
You can the 1.73.0.22306 build to work on Windows 7 if you edit resources/app/out/vs/code/electron-main/main.js at position :82:51909 to replace `hostname:(0,k.hostname)()` to `hostname:0`.

**Comment by Rike-cz** [comment]
Will be happy to have VSCodium on Windows 7 too. [For Microsoft it isn't functional decision, but the product one (whatever it means)](https://github.com/microsoft/vscode/issues/162524#issuecomment-1287079046). Electron looks to be fixed.

**Comment by LexaGV** [comment]
It's extremely ridiculous to see "no support for Win7" in Electron, which has PRIMARY GOAL to be "multiplatform environment" to run applications! :) What a stupid logic! "We support Linux, because we are multiplatform, but we drop support for Win7, where we originally ALWAYS WORKED, because some stupid put in sources "Win10 only" code". Viva! :)
I wish to hear what Electron needs from Win10 and what doesn't exist in Win7 that they drop support for Win7.

**Comment by GitMensch** [comment]
As noted above: Electron needs Windows 10 to _build_ for Windows, because its build tools only run there; but with its September release (19.1.0) _runs_ again on Windows 7 (not older as the used functions are just not available there). Microsoft made the decision to not care about older OS as Windows 10, because the official support for those ended already and they don't want to publish executables working there.
The _idea_ is to now use an updated version of Electron with "Windows 7 support back again" - at least with the insider version - and see where this leads.

If you could help with that by doing so _locally_ (on a Windows10/11 machine, of course) and test and report back, that would be very useful, otherwise you'd need to wait until someone finds the time to do so (or pay a dev to do this [no, I don't have any time available and I've never built vscodium my self so far]).

**Comment by lorand-horvath** [comment]
The latest VSCode Insiders 1.74.0 runs fine on Windows 7 if you start it from the zip file. The exe build doesn't install on Win 7 since they blocked the installer.
https://code.visualstudio.com/docs/?dv=winzip&build=insiders

**Comment by GitMensch** [comment]
Interesting, then we should get that "for free" with the vscodium insider and likely both vscodium and vscode 1.74, too.
I guess they mainly increased the electron dependency.

As vscodium doesn't block the installer (as far as I know, in the unlikely case this is done then this will be dropped) we should have Windows 7 installers working here with both the insider and later the release, too!

... the next thing would be getting the insider versions to build correct again https://github.com/VSCodium/vscodium/actions.

**Comment by lorand-horvath** [comment]
Yes, VSCode 1.74 Insiders bumped electron from 19.0.17 to 19.1.3, which includes the fix for Windows 7, see https://github.com/electron/electron/pull/35702

**Comment by solmersa** [comment]
I can confirm the fix proposed by @AXKuhta works fine on Windows 7 SP1.  I am running VSCodium 1.73.1.22314.

> You can get the 1.73.0.22306 build to work on Windows 7 if you edit resources/app/out/vs/code/electron-main/main.js at position :82:51909 to replace `hostname:(0,k.hostname)()` to `hostname:0`.

**Comment by GitMensch** [comment]
@solmersa Thanks for confirming that. Some questions: Does the installer work fine on Win7? Did you only started it or already verified that editing, possibly debugging, extensions and task work fine?

So we have the insider version running on Win7 "by default" - as soon as the building works again, currently has [an issue with a patch](https://github.com/VSCodium/vscodium/actions/runs/3434666516/jobs/5726194121#step:9:76); we will also have the stable version running on Win7 once 1.74 is out.

For the current available stable 1.73 (and the currently outdated->older insider version) Win7 support can be restored by adjusting that single piece of code after installation, before the first startup.

**Comment by solmersa** [comment]
@GitMensch, I tested the Python Extension, I edited, debugged with no problem, I used the zip version, no installer.

Let me know if I can assist you testing the 1.74 version.

**Comment by daiyam** [comment]
@GitMensch The insiders version has been patched yesterday and has been generated this morning.
But the `code.iss` (which generated the `.exe`) is requiring at least Windows 8 (https://github.com/microsoft/vscode/issues/162524#issuecomment-1272794183). That can be patched.

**Comment by GitMensch** [comment]
@solmersa: thank you for this verification and fast response

@daiyam: the insider version was not available when writing my post, but I see it is now. Thanks.
Can you please patch the installer (also for the stable version)? I think we can close this issue afterwards (even if there's still an issue with the stable version, but even this could be fixed by adding a small `sed -i s'hostname:(0,k.hostname)()/hostname:0/' resources/app/out/vs/code/electron-main/main.js` to the build scripts and re-trigger a stable build).

**Comment by lorand-horvath** [comment]
The latest VSCode 1.74 Insiders just came out, the zip build works fine on Win 7 x64 - VSCodium should as well.

![image](https://user-images.githubusercontent.com/72015221/201317255-c7afb4f4-3ecb-400c-bb55-c55ee79d442b.png)

**Comment by daiyam** [comment]
I won't do `sed -i s'hostname:(0,k.hostname)()/hostname:0/' resources/app/out/vs/code/electron-main/main.js` since it might break something else on Windows 7 or other platforms. I will patch the `code.iss` so that the `.exe` work with Win7.

**Comment by GitMensch** [comment]
Thanks for checking and for the upcoming code.is patch.

We'll just keep this issue open until, presumably next month, 1.74 is out; letting people know how they _likely_ can get 1.73 to start under Win7 if they don't want to install the insiders version in the meantime.

**Comment by daiyam** [comment]
PR #1329 done

**Comment by guy-rouillier** [comment]
This issue affects me too.  I just tried VSCodium for the first time after getting repeated messages from VS Code about not supporting Windows 7 anymore.  I was hoping that VSCodium overcame that restriction.  Sorry for this "me too" post, but I don't see any button to click for "this issue affects me too".  I appreciate everyone who is looking into this.

**Comment by daiyam** [comment]
@guy-rouillier The current Insiders version is fixing the problem. You can find the binaries at https://github.com/VSCodium/vscodium-insiders/releases

**Comment by guy-rouillier** [comment]
@daiyam Thank you.  After reading mention of that here, I downloaded the insiders build and tested it. It works on Windows 7 Pro 64-bit.  I'll watch to see if either VS Code or VSCodium incorporates that change and adapt accordingly.  I appreciate you notifying me.

**Comment by GitMensch** [comment]
@guy-rouillier: VS Code explicit dropped Win7 support, it won't come back. VSCodium on the other side explicit added it, so as soon as vscode 1.74 is released - which [is planned early December](https://github.com/microsoft/vscode/issues/165643) - VSCodium release will have that. And if you don't want to use VSCodium insiders in the meanwhile you can download the .zip version of VSCodium, unpack it and make [that single change mentioned above](https://github.com/VSCodium/vscodium/issues/1264#issuecomment-1311011830) before starting it to have that with VSCodium 1.73.

**Comment by guy-rouillier** [comment]
@GitMensch Thanks.  I'll work with the VS Code Insiders build for the next couple weeks until the December release.

**Comment by daiyam** [comment]
@GitMensch The .exe should also work since I've updated their requirement to include win7

**Comment by moon91210** [comment]
@daiyam I just tried the last 4 insider builds, I tried the msi, exe and zip versions. None of them work. The editor just doesn't show up at all, just like MS's vscode. I am on Win7 Pro x64.

**Comment by daiyam** [comment]
@moon91210 Yeah, those build are broken on all platforms... I have to push a newer version, I will try tonight. I have to re-update the patches...

**Comment by GitMensch** [comment]
.. and just after a working version again we now know that Windows 7 support will go out of scope because Electron follows Google as its base Chromium dropps Win7 support with Electron 23: https://www.electronjs.org/blog/windows-7-to-8-1-deprecation-notice.

> # Deprecation timeline
> The following is our planned deprecation timeline:
>
>* November 2022: [Electron 22 is released](https://www.electronjs.org/blog/electron-22-0).
>* December 2022: The Electron team is entering a quiet period for the holidays
>* January 2023: Windows 7 & 8 related issues are accepted for all supported release branches.
>* February 7 2023: Electron 23 is released.
>* February 8 2023 - May 29 2023: Electron will continue to accept fixes for supported lines older than Electron 23.
>* May 30 2023: Electron 22 reaches the end of its support cycle.

The questions:
* Do we already know when vscode is planning to switch to Electron 22 (which has Win7 Support until end of May 2023)?
   I've seen https://github.com/microsoft/vscode/issues/166265 which is also referenced as "investigating switch to that" for vscode 1.74 (not sure what that means for the release cycle as 1.73 was Electron 19).
* Do we already know when vscode is planning to switch to Electron 23?

As soon as the later is done I see no reasonable way to keep Win7 (or 8/8.1) support.

**Comment by lorand-horvath** [comment]
VSCode 1.74 will be released in a few days and will most probably be Electron 19.1 under the hood (Insiders is currently using 19.1.8) and they have been investigating updating to Electron 22, so I guess next year they will make the switch. I wonder why the jump from 19 to 22 directly, which is not what they have been doing historically - a quite carefully planned step-by-step Electron version update has been done until 19. Going directly to 22 is a bit unusual for the team.
Anyway, this means that Electron 22 with Win 7 support will most probably be used in VSCode until approx. 2023 May-June (v1.79 - 1.80).

**Comment by daiyam** [comment]
Well, my main driver isn't supported by Electron 20 so you are kinda lucky :wink:
Maybe, we should add a link to this thread on the readme?

**Comment by GitMensch** [comment]
> Well, my main driver isn't supported by Electron 20 so you are kinda lucky 😉

Note sure what this means - a possible co-release of vscodium with downgraded Electron (which would likely means quite some patches over time as well as some features missing)?

> Maybe, we should add a link to this thread on the readme?

That's too long. What about a _new_ issue that is "locked" titled "VSCodium Support for Windows 7" which has a link to this discussion thread and a summary of what we know/plan and adjusting the entry in the welcome page to the new one along with adding a link in the README to the new one?

**Comment by moon91210** [comment]
Why not keep Win 7 support as long as you can? A lot of people choose VSCodium on purpose to get away from Microsoft. Win 7 is the only MS OS where you can properly disable all telemetry & automatic updates. I would switch to Linux but with Win7 I can still run my favorite applications and games while maintaining my privacy, which I can't do on Linux or with very limited support. Now we're kind of being forced to switch but we don't have to as long as app developers continue to support Win 7.

**Comment by GitMensch** [comment]
As noted above: the main issue is the dependency chain: VSCodium builds from vscode source, vscode has a dependency on Electron, and Electron a dependency on Chromium. So as soon as vscode _requires_ an electron version that is based on a chromium version that does not run on Win 7 any more, VSCodium cannot provide that.
I've outline above what we know about the deprecation (Chromium and Electron), it now all depends on when vscode requires that Electron version 23; which will be earliest February 2023 and likely latest June 2023.

**Comment by LexaGV** [comment]
Ridiculous situation: instead of making native app, we fall into browser as execution environment (in the hope **editor will work everywhere**). And now we decline MILLIONS of Win7 users just because.... browser doesn't run on Win7 ??? What a shame and circus! **What exactly** chromium does on Win10 what cannot be done on Win7 ? Maybe we should STOP using google's sh%%t and stay on the last Win7-compatible Chromium? No need to follow stupids if you can just stay on place.

**Comment by GitMensch** [comment]
That software features vscode via electron, so I won't consider it "toilet software". As we're "downstream" there is nothing we can do without either breaking the software or patch more and more code, which isn't reasonable.

So "end of this line": MS's VSCode does not support Win7 since some versions, VSCodium does so and will support it at least until February 2023 - and otherwise "as long as vscode does not switch to Electron 23".
Then someone will be free to create a PR with a patchset that can be used to downgrade that dependency as an extra set of Windows binaries - if it works and people keep adjusting the patches we can keep a "semi-supported" version of VSCodium on Win7, if no one does the work or it ... doesn't work then you can still use "the then last" VSCodium with Win7 support on your Win7 machine programming for Win7 (I guess that's the same for a bunch of applications).

**Comment by daiyam** [comment]
@moon91210 @LexaGV I understand your situation but the compatible versions will still work in the future. If you want to maintain the compatibility with PR, you will be free to do it.

@GitMensch I agree with you.

**Comment by daiyam** [comment]
> > Well, my main driver isn't supported by Electron 20 so you are kinda lucky 😉
> 
> Note sure what this means - a possible co-release of vscodium with downgraded Electron (which would likely means quite some patches over time as well as some features missing)?

No, I don't care. I use an old ass macOS which I will need to upgrade at one point in time.

**Comment by Rike-cz** [comment]
> **What exactly** chromium does on Win10 what cannot be done on Win7 ?

I would be happy to know this too. It's only "web browser" core. Does anybody know what feature in Windows 10+ is must-to-have? The same browser can work on 20 yrs old scrap with some minimalistic Linux distro, so we are back in circle. Hardware is ok. Microsoft's decision is fault.

**Comment by zedayseekay** [comment]
> > **What exactly** chromium does on Win10 what cannot be done on Win7 ?
> 
> I would be happy to know this too. It's only "web browser" core. Does anybody know what feature in Windows 10+ is must-to-have? The same browser can work on 20 yrs old scrap with some minimalistic Linux distro, so we are back in circle. Hardware is ok. Microsoft's decision is fault.

This change does not withdraw functionality. It withdraws support.

If you wish to continue using the software on "20 years old scrap with minimalistic Linux distro" you can; the only different here is that if you find an issue specific to that equipment, you will not be able to ask for time and money from this project to support it. For 20 year old equipment that seems completely reasonable.

**Comment by Rike-cz** [comment]
@zedayseekay I wanna tell that you are able to run newest Electron with scrap when using Linux, but not with good hardware when using Windows 7 (and cannot upgrade to Windows 10).

**Comment by zedayseekay** [comment]
> @zedayseekay I wanna tell that you are able to run newest Electron with scrap when using Linux, but not with good hardware when using Windows 7 (and cannot upgrade to Windows 10).

...except the Windows 7 issues that prompted this whole thread are now fixed at the effort of the upstream Electron project... so you can.

**Comment by Rike-cz** [comment]
@zedayseekay Sure. Until approx. 2023 May-June as written above.

**Comment by overlisting** [comment]
Why not go with this all the way to Windows NT 3.1?

**Comment by LexaGV** [comment]
@overlisting : Your stupid irony is irrelevant. Win7 was and is the last adequate Windows version. If you do not understand it, sorry, IT is not for you.

**Comment by IntoTheVoid-900** [comment]
I dont use Windows 7, but some of you could try [VxKex](https://github.com/vxiiduu/VxKex) to try and make it run. It is a reverse Windows compatibility tool for running 8+ apps on 7.

**Comment by daiyam** [comment]
I get the same warning for my old macos since 2 months... I've disabled it on my fork.
Should we disable it on VSCodium? @GitMensch  @paulcarroty What are your thought on this?

**Comment by GitMensch** [comment]
As far as I remember that message is actually true - so the warning is not wrong.
While _the old electron version_ does keep support for old Win7 vscode will switch to _new_ electron version that does not work on Win7, isn't this correct? [and out of interest: did anyone checked if that VxKex  helps with that at all]?

Of course it is bad to be seen over and over again, but ideally this problem should be handled upstream (changed to be configurable for example); @daiyam I guess we only can leave "as is" or "remove permanently", correct? In any case, would you mind linking the change that patches this warning away?

**Comment by daiyam** [comment]
> As far as I remember that message is actually true - so the warning is not wrong.

Yes

> Of course it is bad to be seen over and over again

Ya, it's always there... 

>  but ideally this problem should be handled upstream

I don't think they care or that wouldn't have been an issue

> we only can leave "as is" or "remove permanently", correct?

That's why I left it as is. Maybe, it can be patched to not be permanent but not sure.

**Comment by paulcarroty** [comment]
I vote to patch it if it's easy.

**Comment by lorand-horvath** [comment]
The EOL banner was added in commit https://github.com/microsoft/vscode/commit/d186a2b6e608f18370bfa631edd949bd52d00b9b#commitcomment-105858031
and the related PR is https://github.com/microsoft/vscode/pull/177668#discussion_r1147238329

Quoting myself:
`@deepak1556 Please add a neverShowAgain option to the windowseol bannerService in line 721. It's annoying to have to close it every time I start Insiders. I know I'm running an unsupported Windows version and I don't really need a reminder every time. I want to be able to dismiss it forever. Is that possible to do?`

I didn't receive any response, but it's obviously patchable - I've done it locally by removing 
```
this.bannerService.show({
	id: 'windowseol.banner',
	message,
	ariaLabel: localize('windowseolarialabel', "{0}. Use navigation keys to access banner actions.", message),
	actions,
	icon: Codicon.warning
});
```

**Comment by daiyam** [comment]
Here a screen:
![Screen Shot 2023-04-04 at 8 05 15 PM](https://user-images.githubusercontent.com/587742/229879756-1dc7bd23-5306-4fff-a11b-3469f6a008ba.png)

The notification can be disabled ("Don't Show Again") but the banner will always reopen.

The banner can be simply removed or the option `neverShowAgain` will need to be implemented. What do you think?
Adding the option doesn't seem to be complicated but seems a bit unnecessary...

**Comment by cyberhardt** [comment]
great discussion. I'd also like to see that warning removed permanently - it keeps popping up. regarding what some others have said, most issues regarding windows 7 compatibility are usually very minor, like a single line of code in a dll that purposefully breaks on versions older than 10. either that, or it's simply blocked on the installer: sometimes you can install ithe program manually by extracting the file yourself. it baffles me how they so insistently try to kill off w7.

**Comment by GitMensch** [comment]
> it baffles me how they so insistently try to kill off w7

_possibly_ the same reason why they don't want you to run GNU/Linux (less control, less "connected")... and _likely_ because it is, security wise, not that good (obviously switching to GNU/Linux would be much more reasonable, but then there's less to sell for most people there...)

**Comment by LexaGV** [comment]
@cyberhardt "it baffles me how they so insistently try to kill off w7"

Answer is obvious: Win7 doesn't have so much spies, embedded in OS. Literally the one thing keeps you connected w MS: windows update (and they can get any info thru this channel). Win10 is full of that so called "telemetry" and it cannot be switched OFF just by checkboxes - it's embedded much wider than we think. That's why I insist in every FOSS soft to NOT drop Win7 support - trust me, soon or later you'll return to this OS. We already have one sample of dictatorship - Apple. Fight with second one - MS.

**Comment by daiyam** [comment]
The new version is available, the banner won't display if you have selected `Don't Show Again` for the notification :wink:


> That's why I insist in every FOSS soft to NOT drop Win7 support

In that aspect, at least VSCode is better that Neovim... I would like to try it (with AstroNvim) but doesn't work...

**Comment by daiyam** [comment]
Visual Studio Code 1.80 drops the support of Windows 8. It's using Electron 22 which support Windows 7 and 8.
I'm editing VSCodium 1.80 to be able install/run on Windows 7/8 but I can't test them...
We will have to drop Windows 7/8 when Visual Studio Code is moving to Electron 23 which drops Windows 7/8.

**Comment by guy-rouillier** [comment]
Visual Studio Code stopped supplying updates for Windows 7 at release 
1.7.  However, I'm currently running 1.77.0 on Windows 7.  All I had to 
was download the zip file instead of the installer.  Every time I run 
it, it displays a green bar telling me to "Consider upgrading your 
Windows version".  But everything works.

--
Guy Rouillier

**Comment by lorand-horvath** [comment]
The last VSCode (zip) and VSCodium still running on Windows 7 is 1.81.1
The next version 1.82 updates Electron from 22 to 25 and won't run on Win 7 anymore.

**Comment by daiyam** [comment]
I can confirm that 1.82 is using Electron 25 so VSCodium will drop the support for Win7.

**Comment by moon91210** [comment]
@daiyam Why?

![image](https://github.com/VSCodium/vscodium/assets/113142507/950e70e8-fe9b-4f4e-bdf7-7e92e5725615)

I don't want to push you guys but is it really That big of a deal? I really like VSCodium and would like to see Win7 support continued. I think many others users will agree with me who are still on Win7 or have switched back to it for different reasons, no telemetry, ease of use, etc. and prefer VSCodium over VSCode.
Isn't that the whole point of VSCodium? To have no Microsoft telemetry? You only really get that on Windows 7 (or Linux), on Windows 10 there is still a ton of it that is near impossible to disable even with using privacy tools. And don't even get me started on Windows 11.

**Comment by lorand-horvath** [comment]
@moon91210 Electron 23+ includes Chromium 110+ which simply doesn't work on Win 7 anymore. The last Chromium and Chromium-based browsers (e.g. Chrome, Brave, Edge) that work on Windows 7 is version 109.

So any app built with Electron 23 or newer won't work on Win 7, simple as that. It is unfortunate for all of us, I know.

**Comment by GitMensch** [comment]
As noted: the upstream (electron) of the upstream (vscode) uses an upstream (chromium) that doesn't run on windows 7 and more.

Part of the reason is that all of those stopped running tests on operating systems that don't have much of their user base / target any longer and instead put the resources into what they find relevant.

There is zero that VSCodium can do about that.

Of course, if someone wants to create a fork that patches electron 23 to use an earlier chromium or patches vscode to use an earlier version of electron, that likely will be possible - but harder with each new vscode version and already a bunch of work in the beginning.

If you want to stick with Windows 7, then just install the last working VSCodium version from the releases and disable auto update. This of course means that you do not only not get new vscode features and bug fixes but also that several extensions won't update and more over time (because of the vscode minimal version dependency).


On the other hand: all on those points mentioned are of course available with switching to a free and user respecting operating system, which still allows you to use newest versions of vscode and other software.
That's your choice as a user.

I think it is time to close this issue soon, because we cannot do anything to extend the support more.

**Comment by moon91210** [comment]
Thank you guys for the explanation. It make sense to me now and I've got that last working version installed without updates. Really appreciate you for having supported Win7 up to this point.

**Comment by lorand-horvath** [comment]
I'll just leave this here: https://github.com/microsoft/vscode/issues/184021#issuecomment-1719102800

**Comment by cyberhardt** [comment]
> I'll just leave this here: [microsoft/vscode#184021 (comment)](https://github.com/microsoft/vscode/issues/184021#issuecomment-1719102800)

you're right. some people around here just have this obsession with having the newest and the shiniest thing, even if it means having a worse experience. sometimes I ask myself if at least a fraction of them are paid by big tech but then I remember there are those so desperate for github's green squares that they'll bump everything they can under the sun if that means having a greener contribution activity history

**Comment by daiyam** [comment]
> some people around here just have this obsession with having the newest and the shiniest thing

Why no sticking to an older/working version of VSCodium/VSCode ?

PS: My machine is running an i5-4570 and I'm still using an iPhone 5S.

**Comment by cyberhardt** [comment]
> Why no sticking to an older/working version of VSCodium/VSCode ?
> 
> PS: My machine is running an i5-4570 and I'm still using an iPhone 5S.

it's what I plan on doing, but I'm pretty sure newer VSCodium and electron versions will still be usable on windows 7/8.1, either due to someone forking vscodium with an older electron build or projects like vxkex enabling it to work on windows 7

**Comment by GitMensch** [comment]
So @cyberhardt @lorand-horvath Can you test and drop a hint if the current zip file VSCodium works with [VxKex](https://github.com/vxiiduu/VxKex) on Win7?

**Comment by guy-rouillier** [comment]
> Why no sticking to an older/working version of VSCodium/VSCode ?

For some people - myself included - VSCode is just a tool.  The code I develop with it is my true objective.  So, I don't study the details of each VSCode release.  I figure there must be something useful in it, otherwise why would the author bother?  So, because I'm aware of the termination of support for Windows 7, which I use on my primary system, I'll just skim the release notes for any warning about Windows 7, and if don't quickly notice anything, I'll just apply the update and get back to working on my code.

With such more malware in the world, we've all been conditioned to apply available updates.

**Comment by cyberhardt** [comment]
> So @cyberhardt @lorand-horvath Can you test and drop a hint if the current zip file VSCodium works with [VxKex](https://github.com/vxiiduu/VxKex) on Win7?

I'm not home for the week so I can't test it on my machine

**Comment by Selva124** [comment]
it still doesnt work it shows error kernel32

**Comment by Selva124** [comment]
whats the latest version of vscodium that supports win 7?

**Comment by GitMensch** [comment]
> whats the latest version of vscodium that supports win 7?

That is noted less than 20 issues above - it is 1.81.x and the newest VSCodium release that matches is [1.81.1](https://github.com/VSCodium/vscodium/releases/tag/1.81.1.23222).

The only "open" question is: do newer versions work with VxKex, but other than this issue we have nothing left here, because as explained above there is nothing that VSCodium will do about replacing one of vscode's main dependencies (with the need to provide bigger and bigger patches).

If you have news about "it works when I do X, even when staying on Windows 7" then please post that, otherwise please don't post to this issue.

**Comment by lorand-horvath** [comment]
@GitMensch Why are you never actually linking to my comment? (You've done this 3 times AFAICT, which is a bit weird)
"This is noted" or "as noted above" is usually not enough. Also re-stating the same thing again without linking to your source is not good practice in general. If you'd proceed as such in peer-reviewed papers, for example, a lot of trouble would ensue.

**Comment by GitMensch** [comment]
> The last VSCode (zip) and VSCodium still running on Windows 7 is 1.81.1 The next version 1.82 updates Electron from 22 to 25 and won't run on Win 7 anymore.

.... as noted by @lorand-horvath on August 20 at https://github.com/VSCodium/vscodium/issues/1264#issuecomment-1685223700 ...

which is kind of a follow-up to what @daiyam wrote in https://github.com/VSCodium/vscodium/issues/1264#issuecomment-1624239124 on July 6

> Visual Studio Code 1.80 drops the support of Windows 8. It's using Electron 22 which support Windows 7 and 8. I'm editing VSCodium 1.80 to be able install/run on Windows 7/8 but I can't test them... We will have to drop Windows 7/8 when Visual Studio Code is moving to Electron 23 which drops Windows 7/8.

**Comment by AXKuhta** [comment]
This fork https://github.com/Alex313031/codium/releases will continue to ship builds of vscodium with downgraded electron [until January of next year](https://github.com/Alex313031/codium/releases/tag/1.81.1.23238).

**Comment by GitMensch** [comment]
> This fork https://github.com/Alex313031/codium/releases will continue to ship builds of vscodium with downgraded electron until January of next year.

Thank you! Would you mind editing its README so this gets clearer and the "downgraded electron until January 2024" is also included? What will happen afterwards (and what's the reason to extend for Win7 for [only] 4 months)?

**Comment by AXKuhta** [comment]
@GitMensch I'm not the author of this fork! I just wanted to point out that it exists. No idea as to why they only plan to ship with downgraded electron for 4 more months.

@Alex313031 could you take a look at this?

**Comment by Alex313031** [comment]
@GitMensch @AXKuhta @LexaGV @guy-rouillier @Selva124 I need to update that. I have many Electron projects, including Codium. I was originally only going to use Electron 22 until October, but now I am going to use it longer than that, probably until October of 2024, when ESU for Windows 7 ends.

So rest assured, Codium, as well as all my other Electron projects, will continue to run on Win 7/8/8.1.

Also, I have a fork of upstream VSCode that similarly runs on Win7 > https://github.com/Alex313031/vscode-win7
My other Electron projects :
https://github.com/Alex313031/beaker-ng
https://github.com/Alex313031/quark-player
https://github.com/Alex313031/universal-paperclips-electron

**Comment by spacedrone404** [comment]
> @GitMensch @AXKuhta @LexaGV @guy-rouillier @Selva124 I need to update that. I have many Electron projects, including Codium. I was originally only going to use Electron 22 until October, but now I am going to use it longer than that, probably until October of 2024, when ESU for Windows 7 ends.
> 
> So rest assured, Codium, as well as all my other Electron projects, will continue to run on Win 7/8/8.1.
> 
> Also, I have a fork of upstream VSCode that similarly runs on Win7 > https://github.com/Alex313031/vscode-win7 My other Electron projects : https://github.com/Alex313031/beaker-ng https://github.com/Alex313031/quark-player https://github.com/Alex313031/universal-paperclips-electron

Is there any way to get Win7 builds of VS code?
I can't find them on project page.

**Comment by GitMensch** [comment]
So far that seems to be only https://github.com/Alex313031/codium/releases (note @Alex313031 I do wonder if there are CPU "limits" like with your Thorium).

**Comment by spacedrone404** [comment]
> So far that seems to be only https://github.com/Alex313031/codium/releases (note @Alex313031 I do wonder if there are CPU "limits" like with your Thorium).
So, no original VS code at this time..it's a pity.
A what is with cpu limits?

**Comment by badrelmers** [comment]
only codium by Alex313031 worked for me in win 7 thanks to God, @Alex313031 thank you very much for your great work. https://github.com/Alex313031/codium/releases

VxKex did not work at all, vscodium still asked for some dll...

VSCodium 1.81.1 does not run in win 7, vscode 1.81.1 works fine!

to hide the upgrade notice in vscode 1.81.1, i found this trick
![image](https://github.com/VSCodium/vscodium/assets/40046489/36c66f68-f606-41ae-a2b8-0e24ae800ded)

inside VSCode folder find "resources\app\out\vs\workbench\workbench.desktop.main.nls.js"
then change
`"{0} on {1} will soon stop receiving updates. Consider upgrading your windows version.","Learn More"
`
to
`,`

it worked fine, but anyway now that we have codium by Alex313031 we do not need anything else, this is a gem that should be paid with gold.

thank you all for your rich information shared in this thread.

**Comment by spacedrone404** [comment]
Mirror post of above mentioned things on how to keep vscode alive under Win7 environment:
https://trackerninja.codeberg.page/post/how-to-run-latest-vs-code-in-windows-7-x64/

**Comment by GitMensch** [comment]
As noted: the upstream (

Am 11. September 2023 11:39:59 MESZ schrieb null ***@***.***>:

**Comment by GitMensch** [comment]
I think we should close this issue now, potentially by mentioning Alex' binaries that are based in VSCodium in the README. This would then also be the place when we cannot provide binaries for older GNU/Linux environments any more (to provide both he based the builds on an old node version).

For the Windows installers we should ensure that they don't declare Win 7/8 compat.

@daiyam what do you think?

**Comment by daiyam** [comment]
> For the Windows installers we should ensure that they don't declare Win 7/8 compat.

Should already be the case

> potentially by mentioning Alex' binaries

Sorry, I don't trust the guy who hides p*** images in his rebuilds.

**Comment by badrelmers** [comment]
I just found days ago that [story ](https://alex313031.blogspot.com/2024/01/the-good-bad-and-ugly.html).

For me in fact I trust a guy who hides p*** images in his rebuilds more than other anonymous rebuilds, because it seems not to be an organisation but just a simple human like us lol. What scares me the most is when a Chinese, American or Russian organization, for example, releases us a browser that seems so good and in the end it is nothing more than an organization to spy on us, like https://browser.360.cn/ee/en.html I install it on an old XP inside Sandboxie and was the only browser able to solve all the XP problems with the new web world sites. after using it for some months one day Sandboxie started acting differently, giving me some strange errors never seen before and changing the resolution and blocking all access to XP, which is only resolved after restarting the computer every time it happens. until one day I found that some programs were installed inside XP that I have never installed since I never install anything inside a computer except the official things from big organizations like Microsoft, Google, Adobe...etc, when I install something I always do it inside Sandboxie, but this time I found that some programs to manipulate photos and convert audio were installed outside Sandboxie!!, there I knew that this browser hacked Sandboxie and hacked my system, and I have never trusted again even in Sandboxie protection, and even less in the things that shine and that belong to  unknown organizations. so this Alex seems innocent of those things and this is what matters most.

And also posting a p*** photo is not a crime in this liberal world :(, well I am against adult content because I am Muslim and that is against my religion and principles, but blaming a young person for a photo while the large multinational multimillionaire transcontinental companies like Google, Bing, Yandex fill their search engines with adult content I can't blame a young man for posting a photo... and if we are going to blame that young man then we have to blame Microsoft first and stop using Github first for allowing p*** in his search engine lol...

"O you who have believed, be persistently standing firm for Allah, witnesses in justice, and do not let the hatred of a people prevent you from being just. Be just; that is nearer to righteousness. And fear Allah; indeed, Allah is Acquainted with what you do." Quran 5-8

**Comment by i486** [comment]
Latest build [1.87.2.24072](https://github.com/VSCodium/vscodium/releases/tag/1.87.2.24072) works fine with latest [VxKex](https://github.com/vxiiduu/VxKex/releases/tag/Version1.1.1.1375)

**Comment by Alex313031** [comment]
@sergeevabc @daiyam @i486 @paulcarroty @guy-rouillier  @spacedrone404 @GitMensch Yes, my Codium fork is the only up to date VSCode fork for Windows 7 without hacks, kernel extenders, etc. 

Supermium-Electron is released by win32ss, and I made an issue there regarding stable builds.

And @GitMensch all Codium builds are compiled with SSE3 (which is the same as upstream Electron and Chromium), as well as other compiler optimizations. But no, not with AVX or anything, so there are no CPU limits.

**Comment by spacedrone404** [comment]
As i understand 1.88 is not easily possible on Win7.

**Comment by tamr** [comment]
> As i understand 1.88 is not easily possible on Win7.
Why not? You can run with the help of extended kernel https://github.com/vxiiduu/VxKex

**Comment by Alex313031** [comment]
@tamr @spacedrone404 I have 1.89 running natively on Windows 7, without VxKex > https://github.com/Alex313031/codium/releases/tag/1.89.0.24129

**Comment by spacedrone404** [comment]
> I have 1.89 running natively on Windows 7,

Yeah i saw this build of yours. Do you think we can run 2.0 on Windows 7 or it is already near the possible edge?
